src/python_testing/TC_DRLK_2_9.py

```diff
@@ -0,0 +1,566 @@
+            response = await self.send_single_cmd(cmd=drlkcluster.Commands.SetCredential(
+                operationType=operationType,
+                credential=credentials,
+                credentialData=credentialData,
+                userStatus=userStatus,
+                userType=userType,
+                userIndex=userIndex),
+                endpoint=self.app_cluster_endpoint,
+                timedRequestTimeoutMs=1000)
+            asserts.assert_true(type_matches(response, drlkcluster.Commands.SetCredentialResponse),
+                                "Unexpected return type for SetCredential")
+
+            if (statuscode != custom_stautus_code):
+                asserts.assert_true(response.status == statuscode,
+                                    "Error sending SetCredential command, status={}".format(str(response.status)))
+            else:
+                asserts.assert_true(response.status == 2 or response.status == 3,
+                                    "Error sending SetCredential command, status={}".format(str(response.status)))
+        except InteractionModelError as e:
+            logging.exception(e)
+            asserts.assert_equal(e.status, statuscode, f"Unexpected error returned: {e}")
+
+    async def clear_credentials_cmd(self, credential, step=None, expected_status: Status = Status.Success):
+        try:
+
+            await self.send_single_cmd(cmd=drlkcluster.Commands.ClearCredential(credential=credential),
+                                       endpoint=self.app_cluster_endpoint,
+                                       timedRequestTimeoutMs=1000)
+        except InteractionModelError as e:
+            logging.exception(e)
+            asserts.assert_equal(e.status, expected_status, f"Unexpected error returned: {e}")
+
+    @async_test_body
+    async def test_TC_DRLK_2_9(self):
+
+        self.Attrib = 0
+        self.numberofcredentialsupportedperuser = None
+        self.common_cluster_endpoint = 0
+        self.app_cluster_endpoint = 1
+        self.invalid_credential_type = 9
+        self.user_unique_id1 = 6459
+        self.user_unique_id2 = 1111
+        self.user_name = "xxx"
+        self.credentialIndex_1 = 1
+        self.credentialIndex_2 = 2
+        self.credentialIndex_3 = 3
+        self.operationType = drlkcluster.Enums.DataOperationTypeEnum.kAdd
+        self.userIndex_1 = 1
+        self.userIndex_2 = 2
+        self.invalid_user_status = 5
+        self.invalid_user_type = 10
+        self.pics_send_setcredential_pin_cmd = False
+        self.pics_send_setcredential_rfid_cmd = False
+        self.pin_code = b""
+        self.pin_code1 = b""
+        self.pin_code2 = b""
+        self.inavlid_pincode = b""
+        self.rfid_tag = b""
+
+        if self.pics_guard(self.check_pics("DRLK.S.F00") and self.check_pics("DRLK.S.F08")
+                           and self.check_pics("DRLK.S.C22.Rsp") and self.check_pics("DRLK.S.C23.Tx")):
```

Comment:
self.pics_guard must be always used before using self.step do not define it will print misleading messages in terminal